src/main/java/com/limechain/network/protocol/grandpa/messages/neighbour/NeighbourMessageScaleReader.java

```
@@ -0,0 +1,22 @@
+package com.limechain.network.protocol.grandpa.messages.neighbour;
+
+import com.limechain.network.protocol.grandpa.messages.GrandpaMessageType;
+import io.emeraldpay.polkaj.scale.ScaleCodecReader;
+import io.emeraldpay.polkaj.scale.ScaleReader;
+import io.emeraldpay.polkaj.scale.reader.UInt64Reader;
+
+public class NeighbourMessageScaleReader implements ScaleReader<NeighbourMessage> {
+    @Override
+    public NeighbourMessage read(ScaleCodecReader reader) {
+        int messageType = reader.readByte();
+        if (messageType != GrandpaMessageType.NEIGHBOUR.getType()) {
+            return null;
```

Maybe throw an error?

---

Done.